cmake/bob_robotics.cmake

```diff
@@ -365,17 +365,24 @@ macro(BoB_build)
         set(CMAKE_EXE_LINKER_FLAGS "-Wl,--allow-multiple-definition")
     endif()
 
-    # Use C++14. On Ubuntu 16.04, seemingly setting CMAKE_CXX_STANDARD doesn't
+    # If C++ standard has not been specified explicitly either in a
+    # CMakeLists.txt file or on the command line, set the standard to C++14, the
+    # minimum supported by BoB robotics.
+    if(NOT CMAKE_CXX_STANDARD)
+        set(CMAKE_CXX_STANDARD 17)
+    endif()
+    set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+    # On Ubuntu 16.04, seemingly setting CMAKE_CXX_STANDARD by itself doesn't
     # work, so add the compiler flag manually.
     #
     # Conversely, only setting the compiler flag means that the surveyor example
     # mysteriously gets linker errors on Ubuntu 18.04 and my Arch Linux machine.
     #       - AD
     if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
```

Comment:
does this get set for clang as well? I know in some contexts e.g. #defining __GNUC__ clang pretends to be GCC but not sure about this one

Reply:
No, this'll only work for gcc. It *might* not be an issue depending on whether or not clang also needs this hack in order to set the c++ standard correctly, but I've now also enabled it for clang for consistency.